docs/development-instructions.rst

```diff
@@ -1,30 +1,139 @@
 Development Instructions
-===========================
+=========================
 
-1. Clone the `ArduinoCore-API <https://github.com/arduino/ArduinoCore-API>`_ repo to any convenient location::
+..
+   TODOS:
+   - Contribution guidelines (move here from Contributing.md)
+   - Code conventions
+   - ....
+   - Tools installation:
+      - uncrustify
+      - python ?
+      - pip ?
+      - pre-commit hook?
+      - spellchecker
+      - ... 
+   - Installation the Arduino PSoC6 core
+   - Setting up the development environment
 
-   git clone https://github.com/arduino/ArduinoCore-API.git
+Environment setup
+------------------
 
-2. Locate in your computer where the Arduino15 packages are installed. 
-   e.g. Windows: c:/Users/%USERNAME%/AppData/local/Arduino15
-        Linux:   ~/.arduino15/
+.. note::
+   | The development environment setup is (currently) only supported on **Linux** |:penguin:|. 
+   | As many tools and scripts are not cross-platform, some of then will not directly work on Windows |:abcd:| or macOS |:apple:|.
 
-3. Change directories::
 
-   cd packages/Infineon-psoc/hardware/psoc/0.0.1
+0. Obviously |:neutral_face:|, install `Arduino IDE (2.0 or higher) <https://docs.arduino.cc/software/ide-v2/tutorials/getting-started/ide-v2-downloading-and-installing/>`_ or `Arduino CLI (1.0.0 or higher) <https://arduino.github.io/arduino-cli/0.24/installation/>`_.
 
-4. Clone this repo::
+1. Install the :ref:`Infineon PSoC™ microcontroller <psoc_core_installation>` boards package.
 
-   git clone https://github.com/Infineon/arduino-core-psoc.git
+..
+   Comment: there is another option to this, which is the manual setup:
+      - Creating the "infineon/hardware/psoc" manually
+      - Creating the "infineon/tools" and manually copy there the MTB tools pacakages.
 
-5. Change directories::
+   Ramya: I think we can skip this one, as the contributors should start with the .json package index installation available or?
```

Comment:
Yes we can skip this. I mainly added this step for us. But not needed anymore. We should start with package index installation.

Reply:
Perfect. Technically, we could even have a script that also take cares of all these steps. Like quick-start-dev: Installing arduino-cli if not available, check the path is the default installation path, then clone the repo in the right... then run the git submodules + sym links. 
But that would be maybe a improvement for later or?

Reply:
Comment removed.